### PR TITLE
Fix Google Cloud authentication logic

### PR DESCRIPTION
**Pull request by kyontan**
Followed guides on:
- https://github.com/google-github-actions/setup-gcloud
- https://github.com/google-github-actions/auth

~Also you need to set repository secret `GCP_SERVICE_ACCOUNT_JSON` that contains one-line JSON string of GCP service account key.~

### CODE REVIEW

**Comment by kyontan** [comment]
@hirosassa I think using WIF will provide us better security, and we can avoid managing keys by setting up WIF.
In this PR, I tried to achieve migrating exsting usage of `service_account_key` easily, so that WIF is out of scope.

**Comment by yuzutas0** [comment]
we need to change readme.md.

File: .github/workflows/actions.yml [hunk 1]
```diff
@@ -8,13 +8,13 @@ jobs:
...
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
-      - name: Set up Cloud SDK
-        uses: google-github-actions/setup-gcloud@master
+        uses: actions/checkout@v3
+      - name: 'Authenticate Google Cloud with Service Account'
+        uses: google-github-actions/auth@v1
         with:
-          project_id: ${{ secrets.GCP_PROJECT_ID }}
```
Comment (yuzutas0) [inline .github/workflows/actions.yml:15]: BQ users sometimes need `project_id` param.
e.g. there's service account in ETL/ELT GCP project, some tables & views in each microservices GCP project.

File: .github/workflows/actions.yml [hunk 2]
```diff
@@ -8,13 +8,13 @@ jobs:
...
-          service_account_key: ${{ secrets.GCP_SA_KEY }}
-          export_default_credentials: true
+          credentials_json: ${{ secrets.GCP_SERVICE_ACCOUNT_JSON }}
```
Comment (yuzutas0) [inline .github/workflows/actions.yml:15]: I guess `${{ secrets.GCP_SERVICE_ACCOUNT_JSON }}` is almost the same as `${{ secrets.GCP_SA_KEY }}` .
Can we use the same variable name?

Comment (kyontan) [reply .github/workflows/actions.yml]: updated

File: .github/workflows/actions.yml (hunk 1 above)
Comment (kyontan) [reply .github/workflows/actions.yml:15]: updated

**Review by hirosassa — APPROVED** [review]
Looks great!

File: .github/workflows/actions.yml (hunk 1 above)
Comment (yuzutas0) [reply .github/workflows/actions.yml:15]: Looks Good.

File: README.md [hunk 3]
```diff
@@ -10,8 +10,10 @@
     - role/BigQuery.jobUser
   - export: json key file
 - GitHub Actions
-  - secrets.GCP_PROJECT_ID
-  - secrets.GCP_SA_KEY
+  - `secrets.GCP_PROJECT_ID`
+    - Project ID (not project number) of the Google Cloud project
+  - `secrets.GCP_SA_KEY`
+    - Set json key (preferred to compress in single-line)
```
Comment (yuzutas0) [inline README.md:16]: thanks for improvement.

File: .github/workflows/actions.yml (hunk 2 above)
Comment (yuzutas0) [reply .github/workflows/actions.yml]: LGTM

**Review by yuzutas0 — APPROVED** [review]

**Pull request merged**